main.py

```diff
@@ -0,0 +1,176 @@
+        new_values[i] = int(new_values[i])
+    new_values.sort()
+    first_highest_temprature = new_values[-1]
+    second_highest_temprature = new_values[-2]
+
+    new_values1 = [list for key, list in dict2.items()]
+    for i in range(0, len(new_values1)):
+        new_values1[i] = int(new_values1[i])
+
+    new_values1.sort(reverse=True)
+    first_lowest_temprature = new_values1[-1]
+    second_lowest_temprature = new_values1[-2]
+
+    first_second_low_high_temprature(first_highest_temprature, second_highest_temprature,
+                                     first_lowest_temprature, second_lowest_temprature, month, year)
+
+
+def first_second_low_high_temprature(maxT, secondmaxT, minT, secondminT, month, year):
+    print(" ")
+
+    print(month + " " + year)
+
+    print("01 ", end="")
+    for i in range(maxT):
+        print("+", end="")
+    print(" " + str(maxT) + "C")
+
+    print("01 ", end="")
+    for i in range(secondmaxT):
+        print("+", end="")
+    print(" " + str(secondmaxT) + "C")
+
+    print("02 ", end="")
+    for i in range(minT):
+        print("+", end="")
+    print(" " + str(minT) + "C")
+
+    print("02 ", end="")
+    for i in range(secondminT):
+        print("+", end="")
+    print(" " + str(secondminT) + "C")
+
+
+def monthly_average_values(dict, dict1, dict2, ):
+    new_values = [list for key, list in dict.items()]
+    for i in range(0, len(new_values)):
+        new_values[i] = int(new_values[i])
+
+    new_values1 = [list for key, list in dict1.items()]
+    for i in range(0, len(new_values1)):
+        new_values1[i] = int(new_values1[i])
+
+    new_values2 = [list for key, list in dict2.items()]
+    for i in range(0, len(new_values2)):
+        new_values2[i] = int(new_values2[i])
+
+    print("Highest Average: " + str(ceil(mean(new_values))) + "C")
+    print("Lowest Average: " + str(ceil(mean(new_values1))) + "C")
+    print("Average Mean Humidity: " + str(ceil(mean(new_values2))) + "%")
+
+
+def yearly_values(dict1, dict2, dict3):
+    max_temprature = max(dict1.values())
+    key_list = list(dict1.keys())
+    val_list = list(dict1.values())
+
+    # print key with val temp
+    position = val_list.index(max_temprature)
+    print("Highest: " + max_temprature + "C on " + key_list[position])
+
+    min_temprature = max(dict2.values())
+    key_list = list(dict2.keys())
+    val_list = list(dict2.values())
+
+    # print key with val temp
+    position = val_list.index(min_temprature)
+    print("Lowest: " + min_temprature + "C on " + key_list[position])
+
+    max_humidity = max(dict1.values())
+    key_list = list(dict1.keys())
+    val_list = list(dict1.values())
+
+    # print key with val temp
+    position = val_list.index(max_humidity)
+    print("Humidity: " + max_humidity + "% on " + key_list[position])
+
+
+def file_processing(i, files_list):
+    index = i
+    files_list = files_list
+    files_path = "/home/hamza/Downloads/weatherfiles/weatherfiles/"
```

Comment:
We are taking file path as command line argument, what is the point of hardcoding it in code?